pkg/controller/stack/stack_controller.go

```diff
@@ -267,8 +251,9 @@ func (r *ReconcileStack) Reconcile(request reconcile.Request) (reconcile.Result,
 			reqLogger.Error(err, "Failed to update Stack", "Stack.Name", stack.Stack)
 			// Update Stack status with failed state
 			instance.Status.LastUpdate = &pulumiv1alpha1.StackUpdateState{
-				State:     pulumiv1alpha1.FailedStackStateMessage,
-				Permalink: permalink,
+				LastAttemptedCommit: instance.Spec.Commit,
+				State:               pulumiv1alpha1.FailedStackStateMessage,
+				Permalink:           permalink,
```

Comment:
Will this result in overwriting `lastSucceasfulCommit` to nil if it was present?

Reply:
Yes. Thanks for pointing out. Updated now.